components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/dao/impl/AuthorizedAPIDAOImpl.java

```diff
@@ -48,33 +54,7 @@ public class AuthorizedAPIDAOImpl implements AuthorizedAPIDAO {
     public void addAuthorizedAPI(String applicationId, String apiId, String policyId,
                                  List<Scope> scopes, int tenantId) throws IdentityApplicationManagementException {
 
-        try (Connection dbConnection = IdentityDatabaseUtil.getDBConnection(true)) {
-            try {
-                PreparedStatement prepStmt = dbConnection.prepareStatement(ApplicationMgtDBQueries.ADD_AUTHORIZED_API);
-                prepStmt.setString(1, applicationId);
-                prepStmt.setString(2, apiId);
-                prepStmt.setString(3, policyId);
-                prepStmt.execute();
-
-                prepStmt = dbConnection.prepareStatement(ApplicationMgtDBQueries.ADD_AUTHORIZED_SCOPE);
-                boolean isSystemAPI = APIResourceManagementUtil.isSystemAPIByAPIId(apiId);
-                for (Scope scope : scopes) {
-                    prepStmt.setString(1, applicationId);
-                    prepStmt.setString(2, apiId);
-                    prepStmt.setString(3, scope.getName());
-                    prepStmt.setObject(4, isSystemAPI ? null : tenantId);
-                    prepStmt.addBatch();
-                    prepStmt.clearParameters();
-                }
-                prepStmt.executeBatch();
-                IdentityDatabaseUtil.commitTransaction(dbConnection);
-            } catch (SQLException | APIResourceMgtException e) {
-                IdentityDatabaseUtil.rollbackTransaction(dbConnection);
-                throw e;
-            }
-        } catch (SQLException | APIResourceMgtException e) {
-            throw new IdentityApplicationManagementException("Error while adding authorized API.", e);
-        }
+        this.addAuthorizedAPI(applicationId, apiId, policyId, scopes, null, tenantId);
```

Comment:
lets pass empty array

Reply:
Fixed in 2d72c264576eb9e95bb40ef8fef70f8cdd5cbeb1